os/board/Kconfig

```diff
@@ -190,6 +190,37 @@ config BOARD_CRASHDUMP
 		"machine state" in a place where on the next reset can write it
 		to more sophisticated storage in a sane operating environment.
 
+if BOARD_CRASHDUMP
+
+config BOARD_COREDUMP_FLASH
+	bool "Enable corelog dump to flash"
+	default n
+	depends on MTD_PARTITION_NAMES
+	---help---
+		If selected it will dump the assert log information to flash.
+
+config BOARD_RAMDUMP_FLASH
+	bool "Enable ramdump to flash"
+	default n
+	depends on MTD_PARTITION_NAMES
+	---help---
+		If selected it will dump the RAM contents to flash.
+
+config BOARD_RAMDUMP_UART
+	bool "Enable ramdump via UART"
+	default n
+	---help---
+		If selected it will send the RAM dump via UART.
+
+config BOARD_SMARTFS_DUMP
+	bool "Enable SmartFS dump when crashed.
+	default n
+	depends on FS_SMARTFS && MTD_PARTITION_NAMES
+	---help---
+		If selected, smartfs is dumped when the current board is crashed.
```

Comment:
I think dumped result will be saved into smartfs, not dump the smartfs when board is crashed.